### ISSUE TITLE
Offline new moth locations

### ISSUE THREAD

**Issue by JurrienVanDeijk** [BiologicalRecordsCentre/ABLE#540]
@DavidRoy  and @kazlauskis. I was last weekend in the Ardennes, and I have placed several moth traps. However, the connection there is poor, and I wasn't able to make new counting locations because I was most of the time offline. Could it be made possible to also make those new locations when you are offline? Because I think there are more locations in Europe with a poor internet connection

**Comment by kazlauskis** [comment]
Thanks for creating the ticket here. Yes, it shouldn't be too difficult to implement basic fully offline moth trap set-up and surveying.

**Comment by larspett** [comment]
@kazlauskis  I noticed today that the app suggested a trap site quite far away from the actual position despite having good internet connection. Even when asking for repositioning, it remained at the same position. I am not sure about how to deal with this, would I need to reposition the site later? Where would I do that and why is it not updating the position in the app but rather insists on its 1st suggestion?  
![3](https://user-images.githubusercontent.com/19598308/234219781-bbd3d6b1-d0a9-4a26-a2e6-53a987e5edc0.jpg)

**Comment by kazlauskis** [comment]
I've checked your moth trap list and the "Stensoffa" trap looks like it is in the correct position now. 

You can set up new moth traps in the app, but you can only edit those on the website. After a trap was changed on the website, it might take some 15mins for the location to be updated in the reports. The app automatically refreshes the location list in the background so you don't need to do anything.

![Screenshot 2023-04-25 at 14 36 22](https://user-images.githubusercontent.com/7383162/234264738-0e8c3485-0a20-4d59-9ab6-f1ecd1ff0e99.png)

**Comment by larspett** [comment]
Yeah I tried editing it on the website and moved it to the right spot. My problem was that the app selected a spot that was not updated (live) even though I enabled the repositioning crosshair which ought to have moved it to the correct position

**Comment by kazlauskis** [comment]
If you have updated a moth trap on the website **after** you've selected that location for a record in the app, then the app might show the old trap location. This is especially true if you are offline or if 15mins haven't passed since the location was updated.

**Comment by larspett** [comment]
No the problem is that the app refused to reposition the dot that it had added when I created the site even though I activated repositioning. But maybe that is the intended behaviour, i.e. that the position sticks to wherever you are (exactly) when you create a site in the app and that further adjustment in the app will not happen? So if I would create a trap position where I sit now and then walk 100 m away to a different position, I would be unable to convince the app to move the trap position to the new coordinates even with positioning enabled?

**Comment by kazlauskis** [comment]
Correct, when you're creating a new moth trap, it will default the trap's location to your current GPS position. But until you upload that newly created trap to the website (i.e. while it is a draft) then you can still update its location using the map - just tap on a corrected location on the map, and it will adjust it.

**Comment by larspett** [comment]
Then that was what didn't happen. I tried multiple ways of repositioning, including activating the crosshair (which usually repositions), and tapping on the correct position but nothing happened

**Comment by kazlauskis** [comment]
@larspett in that case, please create a new separate issue and fill in as many details as possible to help us reproduce this. Thanks!

**Comment by larspett** [comment]
will do next time I am at that site in case it is signal-dependent
fine to close this

**Comment by larspett** [comment]
@kazlauskis we tried to add a first trap for a new user today and experienced something I have seen before, the trap is created but the app cannot select it. Am not sure if there is a separate issue for this so I add it here. The 4G/5G coverage should be fine and there was no other apparent reason for this to happen. The app version is the latest production one. ![image](https://github.com/user-attachments/assets/9eeafd8b-3534-49b3-8141-cafba5cdef39)
![image](https://github.com/user-attachments/assets/8342e494-d484-463e-b186-6da08dde616d)

**Comment by larspett** [comment]
Ping @anastefan02 who experienced the problem

**Comment by kazlauskis** [comment]
@larspett @anastefan02 let's create a new app issue for this - add as many details as you can, e.g. timestamp, can you reproduce it, does it work with other traps etc.

**Comment by larspett** [comment]
@anastefan02 will create a new issue, she is recreating the error today hopefully. When I saw it before, a month ago or so, I think our impression was that it had something to do with geopositioning rights, that it is possible to access the site when creating the new trap location but that the app can have problems positioning itself relative to the trap location, triggering an error. Possibly privacy rules that should trigger a request for positioning but which doesn't. Let's see.
